src/Microsoft.ML.Transforms/TensorflowTransform.cs

```diff
@@ -0,0 +1,681 @@
+        {
+            Host.AssertValue(ctx);
+
+            _batchSize = ctx.Reader.ReadInt32();
+
+            byte[] data = null;
+            ctx.TryLoadBinaryStream("TFModel", r =>
+            {
+                data = r.ReadByteArray();
+            });
+
+            var graph = new TFGraph();
+            graph.Import(data);
+            _session = new TFSession(graph);
+            _bindings = new Bindings(ctx, Source.Schema, this);
+        }
+
+        public static TensorflowTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            var h = env.Register(RegistrationName);
+            h.CheckValue(ctx, nameof(ctx));
+            h.CheckValue(input, nameof(input));
+            ctx.CheckAtModel(GetVersionInfo());
+            return h.Apply("Loading Model", ch => new TensorflowTransform(h, ctx, input));
+        }
+
+        private TFSession LoadTFSession(string modelFile)
+        {
+            var graph = new TFGraph();
+            graph.Import(File.ReadAllBytes(modelFile), "");
+            return new TFSession(graph);
+        }
+
+        public override void Save(ModelSaveContext ctx)
+        {
+            Host.AssertValue(ctx);
+            ctx.CheckAtModel();
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            ctx.Writer.Write(_batchSize);
+
+            var buffer = new TFBuffer();
+            _session.Graph.ToGraphDef(buffer);
+
+            ctx.SaveBinaryStream("TFModel", w =>
+            {
+                w.WriteByteArray(buffer.ToArray());
+            });
+            _bindings.Save(ctx);
+        }
+
+        public void DisposeTFSession()
+        {
+            _session.CloseSession();
+            _session.DeleteSession();
+        }
+
+        private ValueGetter<T> GetSrcGetter<T>(IRow input, int iinfo, int isrc)
+        {
+            return input.GetGetter<T>(_bindings.Infos[iinfo].SrcIndices[isrc]);
+        }
+
+        private ITensorValueGetter CreateTensorValueGetter<T>(IRow input, ColumnType type, int colIndex, TFShape tfShape)
+        {
+            if (type.IsVector)
+                return new TensorValueGetterVec<T>(input, colIndex, tfShape);
+            else
+                return new TensorValueGetter<T>(input, colIndex);
+        }
+
+        private ITensorValueGetter CreateTensorValueGetterVec(IRow input, TFDataType tfType, ColumnType type, int colIndex, TFShape tfShape)
+        {
+            if(tfType == TFDataType.Float)
+            {
+                return CreateTensorValueGetter<float>(input, type, colIndex, tfShape);
+            }
+
+            if (tfType == TFDataType.Double)
+            {
+                return CreateTensorValueGetter<double>(input, type, colIndex, tfShape);
+            }
+
+            if (tfType == TFDataType.Int32)
+            {
+                return CreateTensorValueGetter<DvInt4>(input, type, colIndex, tfShape);
```

Comment:
>DvInt4 [](start = 47, length = 6)

ML.Net types cannot be passed to TF right now.

Reply:
There is work going on on removing ML.Net types and using pure .net types. That will resolve this issue.

---
In reply to: [210172882](https://github.com/zeahmed/machinelearning/pull/3#discussion_r210172882) [](ancestors = 210172882)

Reply:
As long as this work is not done yet, we should either not support these types, or use the ML.NET types and convert to .net types before passing to TF. this method gets the getters of the inputs, and will throw exception if called with Int64, bool or string.

---
In reply to: [210456041](https://github.com/zeahmed/machinelearning/pull/3#discussion_r210456041) [](ancestors = 210456041,210172882)